tls/s2n_security_policies.c

```diff
@@ -891,6 +900,31 @@ int s2n_config_set_cipher_preferences(struct s2n_config *config, const char *ver
     return 0;
 }
 
+static bool s2n_cipher_suite_uses_chacha20_alg(struct s2n_cipher_suite *cipher_suite) {
+    if (!cipher_suite) {
+        return false;
+    }
+
+    /* If chacha20-poly1305 is available, then we can directly check the record_alg */
+    if (s2n_chacha20_poly1305.is_available()) {
+        return cipher_suite->record_alg && cipher_suite->record_alg->cipher == &s2n_chacha20_poly1305;
+    } 
+
+    /* Else chacha20 is not available. In this case, the cipher_suite's record_alg is never set. This can unintentionally
```

Comment:
I am not following what this comment is trying to explain. Can you make it more clear why we are iterating through all record algs here?

Also nit; you seem to be using chacha20/ChaCha20/chacha20-poly1205 all to refer to the same thing. Can you just pick one to use?